### ISSUE TITLE
ErrImagePull in deploy step when image from private repo

### ISSUE THREAD

**Issue by 4ippers** [flant/werf#1989]
When i try to deploy any project to local cluster i always get ErrImagePull: denied: access forbidden.
But build can do push well in same repo.

I used this guide: https://werf.io/documentation/guides/gitlab_ci_cd_integration.html

So I think problem in kubernetes. 
Can werf pass token/acess when he try to deploy and PullImage?

**Comment by diafour** [comment]
Hello. Yes, it is the Kubernetes thing. You need to use `imagePullSecret` in your resources, or use `serviceAccount` with `imagePullSecret`.

https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-pod-that-uses-your-secret

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account

**Comment by joshuataylor** [comment]
Sorry to necro an issue, but I ran into this and wasted a bit of time trying to figure it out, wait for builds, rebuilds and going crazy.

Double check `werf helm get-namespace` is listed as the correct namespace, I'm running a test cluster and this should be `default` but it was my project name.

If you want to change the namespace, change it in `werf.yaml` ([docs](https://werf.io/documentation/reference/werf_yaml.html)):
```
configVersion: 1
project: foo
deploy:
  namespace: default
```

OR add the regcred to your namespace that the above output gives :).

Hope this helps someone who is also running into this! I ran into this for a job kubernetes type and was throwing me off bigtime.
